sklearn/svm/src/libsvm/libsvm_sparse_helper.c

```diff
@@ -3,6 +3,12 @@
 #include "svm.h"
 #include "_svm_cython_blas_helpers.h"
 
+
+#ifndef max
+    #define max(x, y) (((x) > (y)) ? (x) : (y))
```

Comment:
They are needed for correct operator precedence.

For instance, it is better that
```c
max(1+1, 2+2)
```
gets expended to:
```c
(((1+1) > (2+2)) ? (1+1) : (2+2))
```

Though this is a simple example and this might not cause problem for this macro.

Reply:
As @jjerphan mentioned, they are needed for correct operator precedence.
In case one of `x` or `y` has an operator with less precedence than the relational operators (<, >), the result will not be the expected one.
Though, it is true that in this case, this does not happen and probably will never happen (unless someone calls the macro with something like `max(a&b, c)`).
A simple example that showcases the behavior is:
```C
#include <stdio.h>

#ifndef max
    #define max(x, y) ((x > y) ? x : y)
#endif


int main()
{
    int a, b, c;
    a = 1;
    b = 3;
    c = 2;

    // (((1&3) > (2)) ? (1&3) : (2))
    // x = 1&3
    // y = 2
    // (x > y) ? x : y
    // (1&3 > 2) ? 1&3 : 2
    // (1&1) ? 1&3 : 2
    // 1 ?  1&3 : 2

    printf("1&3 = %d\n", a&b);
    printf("max(1&3, 2) = %d\n", max(a&b, c));

    return 0;
}

```